pytorch_lightning/trainer/trainer.py

```diff
@@ -1555,6 +1555,27 @@ def _call_lightning_module_hook(
 
         return output
 
+    def _call_lightning_datamodule_hook(
+        self,
+        hook_name: str,
+        *args: Any,
+        datamodule: Optional["pl.LightningDataModule"] = None,
```

Comment:
We don't need this argument. It's in `_call_lightning_module_hook` for legacy reasons

```suggestion
```